### PR TITLE
chore(dgraph): updating badger to v3

### PR DESCRIPTION
**Pull request by aman-bansal**
This change is [Reviewable](https://reviewable.io/reviews/dgraph-io/dgraph/7292)

### CODE REVIEW

**Review by danielmai — APPROVED** [review]
> Reviewed 66 of 66 files at r1.
> *[Reviewable](https://reviewable.io/reviews/dgraph-io/dgraph/7292)* status: :shipit: complete! all files reviewed, all discussions resolved (waiting on @manishrjain, @martinmr, @pawanrawal, and @vvbalaji-dgraph)

**Review by danielmai — APPROVED** [review]
:lgtm:

> *[Reviewable](https://reviewable.io/reviews/dgraph-io/dgraph/7292#-:-MQzgBCE3n3sbGHUroU0:bnfp4nl)* status: :shipit: complete! all files reviewed, all discussions resolved (waiting on @manishrjain, @martinmr, @pawanrawal, and @vvbalaji-dgraph)

**Review by danielmai — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/dgraph-io/dgraph/7292)* status: all files reviewed, 1 unresolved discussion (waiting on @aman-bansal, @manishrjain, @martinmr, @pawanrawal, and @vvbalaji-dgraph)

___
*[go.mod, line 71 at r1](https://reviewable.io/reviews/dgraph-io/dgraph/7292#-MQzhQKBC-skFNKwjSRK:-MQzhQKBC-skFNKwjSRL:b-jlnbs4) ([raw file](https://github.com/dgraph-io/dgraph/blob/20fd984bd489202cf10462e124ffc154932f4b93/go.mod#L71)):*
> ```
> 	google.golang.org/grpc v1.27.0
> 	google.golang.org/protobuf v1.25.0 // indirect
> ```
Why are these dependencies upgraded or added?

File: x/metrics.go [hunk 1]
```diff
@@ -320,57 +320,57 @@ func init() {
 // NewBadgerCollector returns a prometheus Collector for Badger metrics from expvar.
 func NewBadgerCollector() prometheus.Collector {
 	return prometheus.NewExpvarCollector(map[string]*prometheus.Desc{
-		"badger_v2_disk_reads_total": prometheus.NewDesc(
+		"badger_v3_disk_reads_total": prometheus.NewDesc(
 			"badger_v2_disk_reads_total",
```
Comment (jarifibrahim) [inline x/metrics.go:324]: Add a comment about why this is v2

**Review by jarifibrahim — APPROVED** [review]

**Review by danielmai — APPROVED** [review]
:lgtm:

> Reviewed 13 of 13 files at r2, 1 of 1 files at r3.
> *[Reviewable](https://reviewable.io/reviews/dgraph-io/dgraph/7292#-:-MR61z83BBlKQNPVAKpI:bnfp4nl)* status: all files reviewed, 2 unresolved discussions (waiting on @aman-bansal, @manishrjain, @martinmr, @pawanrawal, and @vvbalaji-dgraph)

**Pull request merged**